### ISSUE TITLE
There was a syncing problem

### ISSUE THREAD

**Issue by tomholub** [FlowCrypt/flowcrypt-android#657]
I opened the app after some time of inactivity.

Then I opened a message.

![image](https://user-images.githubusercontent.com/6306961/61360726-843d7580-a86e-11e9-885c-8daca9628d04.png)

Retrying with the button didn't help. Had to kill the app.

**Comment by DenBond7** [comment]
@tomholub Unfortunately, I can't reproduce this issue. If you have more details, add them, please. I think it's a rare bug.

**Comment by tomholub** [comment]
I'll see if I can encounter it again

**Comment by tomholub** [comment]
A customer just had this problem too. She did not now how to resolve it (she didn't know or think to kill the app).

What does the retry button do today?

Can you make the retry button for this screen completely disconnect from IMAP and reconnect? Maybe also refetch the token? That could give users a chance to fix it.

**Comment by DenBond7** [comment]
@tomholub 
> What does the retry button do today?

The retry button rerun the current task (if a connection is available the task uses it).

> Can you make the retry button for this screen completely disconnect from IMAP and reconnect? Maybe also refetch the token? That could give users a chance to fix it.

I'll try to do that. I think it'll help us.

**Comment by DenBond7** [comment]
@tomholub 
When you will see this error next time please notice the following info (it is important for debugging):

- Is any connection available? (WiFI, 3G)
- Is access to the Internet available? (you must be able to get info from different sites, for example, www.google.com and etc.) Because we can have a situation when we have WIFI connection but there is no internet access

This error means that we receive [MailConnectException](https://javaee.github.io/javamail/docs/api/com/sun/mail/util/MailConnectException.html)

**Comment by tomholub** [comment]
I suspect maybe it's related to an authentication failure, or a dropped tcp connection with some state left over. Fully renewing auth token + reconnecting should really solve it. I remember my internet access was fine, almost certainly on Wifi.

**Comment by DenBond7** [comment]
Got it. Thanks! I've added some changes. I hope they will resolve this issue.

**Comment by seisvelas** [comment]
There are two support emails reporting this error:

https://mail.google.com/mail/u/5/#inbox/FMfcgxwJXCHHDFLkFbvDQKJBpKQNTxWN

>Updating index very slow, later sync error

And:

https://mail.google.com/mail/u/5/#search/sync/FMfcgxwJXCFwXTFvvPFhBtHlSWrwGZTj
 
> First thinks first. How r u ? Than,
> 
> Whats that ?
> I dont know this error ? How can ı fix that ?
> Thank yu.

**Comment by seisvelas** [comment]
Screenshot:

![screenshot](https://user-images.githubusercontent.com/44826516/89114933-3bd80200-d447-11ea-926b-18e0e2e8dc47.png)

**Comment by seisvelas** [comment]
ref: https://mail.google.com/mail/u/human@flowcrypt.com/#inbox/FMfcgxwKjdvMVZNBWnswFZVfXrhgZjnB

**Comment by seisvelas** [comment]
ref: https://mail.google.com/mail/u/human@flowcrypt.com/#inbox/FMfcgxwKjfDHRpGgRcxtKSzRpfmNXbKr

Hopefully this will improve once https://github.com/FlowCrypt/flowcrypt-android/issues/993 is finished.

Is there anything these users can do in the meantime that is likely to improve the situation for them?

**Comment by DenBond7** [comment]
> Is there anything these users can do in the meantime that is likely to improve the situation for them?

Usually, it means that the app has lost a connection to a remote server. But it seems JavaMail not always can resolve reconnection. Unfortunately, for now, a user should kill the app (from `the settings - app settings - close` or from the `recent apps` window). The current realization doesn't allow me to manage such a situation properly. But after #993 I'll be able to do that.

@seisvelas How often do we receive such feedbacks?

**Comment by seisvelas** [comment]
>@seisvelas How often do we receive such feedbacks?

I've received four in the last week, but prior to that much less often. And one of those users from this week I suspect was having a somewhat different problem (emails taking a long time to sync, but not getting this error message), so perhaps that doesn't count towards this specific issue. 

I'll let users know that killing the app may temporarily abate the problem. Sorry to bug, and thanks for the tip!

**Comment by seisvelas** [comment]
ref: https://mail.google.com/mail/u/human@flowcrypt.com/#inbox/FMfcgxwKjdzpWtDQSDTfGFpWHkKFdCJF

**Comment by DenBond7** [comment]
@seisvelas Please tell me do you receive such feedbacks? It should was fixed by #993

**Comment by seisvelas** [comment]
I haven't received any complaints about this since December 4th, so I think we can safely assume that your fix worked. I'll reopen this if any new support emails come in about this.
